Modules/VAMI/VAMI.psm1

```diff
@@ -655,38 +655,46 @@ Function Get-VAMIUser {
     $userAPI = Get-VAMIServiceAPI -NameFilter "accounts"
     $UserResults = @()
 
-    if (($Name -ne "") -and ($null -ne $Name)) {
-        try {
-            $Users = $UserAPI.get($name)
-        } catch {
-            Write-Error $Error[0].exception.Message
-        }
-    } else {
+    # Get a list of users
+    try {
         $Users = $UserAPI.list()
+    } catch {
+        write-error $_
+    }
+
+    # Apply filtering if Name input is provided
+    if ($Name -ne '' -AND $Name -ne $null) {
+        # For 6.5 API, the username is part of the list returnset; for 6.7/7.x API the value from the list is the username. Because of this we will use an OR filter to account for either case.
+        $Users = $Users | Where-Object {$_.username -eq $name -OR $_.value -eq $name}
     }
+
     if ($Users.status) {
+        # This is for 6.5 API, which has a status property; in newer API response there is an enabled property with values of True/False
         foreach ($User in $Users) {
             $UserString = [pscustomobject] @{
                 User = $User.username
                 Name = $User.fullname
                 Email = $User.email
-                Status = $User.status
+                Enabled = if ($User.status -eq 'enabled' ) { $true } else { $false }
+                LastPasswordChange = 'Unknown'
+                PasswordExpiresAt = 'Unknown'

```

Comment:
If these values are DateTime in the other case maybe $null would be a better default than 'Unknown'.